File: Merge_Two_Sorted_Lists.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    ListNode *mergeTwoLists(ListNode *l1, ListNode *l2) {
        if (l1 == NULL) {
            return l2;
        } else if (l2 == NULL) {
            return l1;
        }

        ListNode *head, *pnt1, *pnt2;
        if (l1->val <= l2->val) {
            head = l1;
            pnt1 = l1->next;
            pnt2 = l2;
        } else {
            head = l2;
            pnt1 = l1;
            pnt2 = l2->next;
        }

        ListNode *curpnt = head;

        while (pnt1 != NULL && pnt2 != NULL) {
            if (pnt1->val <= pnt2->val) {
                curpnt->next = pnt1;
                pnt1 = pnt1->next;
            } else {
                curpnt->next = pnt2;
                pnt2 = pnt2->next;
            }

            curpnt = curpnt->next;
        }

        if (pnt1 != NULL) {
            curpnt->next = pnt1;
        } else if (pnt2 != NULL) {
            curpnt->next = pnt2;
        }

        return head;
    }
};

File: Merge_Two_Sorted_Lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None


class Solution:
    # @param two ListNodes
    # @return a ListNode

    def mergeTwoLists(self, l1, l2):
        if l1 is None:
            return l2
        elif l2 is None:
            return l1

        head, pnt1, pnt2 = (l1, l1.next, l2) if l1.val <= l2.val else (l2, l1, l2.next)
        curpnt = head

        while pnt1 is not None and pnt2 is not None:
            if pnt1.val <= pnt2.val:
                curpnt.next = pnt1
                pnt1 = pnt1.next
            else:
                curpnt.next = pnt2
                pnt2 = pnt2.next
            curpnt = curpnt.next

        if pnt1 is not None:
            curpnt.next = pnt1
        elif pnt2 is not None:
            curpnt.next = pnt2

        return head
